gadget/install/install.go

```diff
@@ -28,23 +28,39 @@ import (
 	"github.com/snapcore/snapd/boot"
 	"github.com/snapcore/snapd/gadget"
 	"github.com/snapcore/snapd/logger"
+	"github.com/snapcore/snapd/osutil/disks"
 	"github.com/snapcore/snapd/secboot"
 	"github.com/snapcore/snapd/timings"
 )
 
-func deviceFromRole(lv *gadget.LaidOutVolume, role string) (device string, err error) {
+// deviceFromSystemSeedRole will identify an ondisk device node which is a disk
+// that has the SystemSeed role for the specified gadget volume on it.
+// It could be made to work for more generic roles, but some roles require
+// special handling like ubuntu-data, where on an encrypted system we really
+// need to look for a partition device node for
+// /dev/disk/by-partlabel/ubuntu-data-enc rather than /dev/.../ubuntu-data,
+// which is how this function currently operates and the logic to know whether
+// the device is an encrypted one or a normal one is not yet implemented, so we
+// artificially limit this function to only find the system seed role
```

Comment:
I think it's ok to leave out the bit about other roles to make the comment less confusing.
```suggestion
```